### PR TITLE
Remove Keycloak references

### PR DESCRIPTION
**Pull request by makambalaji**
### What does this PR do?
https://github.com/eclipse/che/issues/21175

### How to test this PR?

After removing Keycloak references, still the test cases should work. Executed below commands for testing purpose
`
> export TS_SELENIUM_BASE_URL=https://192.168.59.165.nip.io
> export TS_SELENIUM_USERNAME=che@eclipse.org
> export TS_SELENIUM_PASSWORD=admin
> export USERSTORY=VscodeYamlPlugin
> npm run test-plugin`
Attaching Yaml plugin execution logs : [yamlPlugin_ExecutionLogs.txt](https://github.com/eclipse/che/files/8197091/yamlPlugin_ExecutionLogs.txt)

> export USERSTORY=Go
> npm run test-all-devfiles
Execution logs for Go spec file:  [Go_spec_execution_logs.txt](https://github.com/eclipse/che/files/8197055/Go_spec_execution_logs.txt)

#### Verify the downstream code with upstream

> 

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: tests/e2e/driver/CheReporter.ts [hunk 1]
```diff
@@ -86,8 +86,8 @@ class CheReporter extends mocha.reporters.Spec {
       if (TestConstants.TS_SELENIUM_RESPONSE_INTERCEPTOR) {
         CheApiRequestHandler.enableResponseInterceptor();
       }
-      await preferencesHandler.setConfirmExit(AskForConfirmationType.never);
-      await preferencesHandler.setTerminalType(TerminalRendererType.dom);
+      // await preferencesHandler.setConfirmExit(AskForConfirmationType.never);
```
Comment (ScrewTSW) [inline tests/e2e/driver/CheReporter.ts:89]: @makambalaji  Why were the preferences handler calls removed? It doesn't contact keycloak, that's purely a Theia thing.

**Comment by makambalaji** [comment]
Looks like iframe tag is removed in the Eclipse Che 7.43 version, So created a separate function "verifyAndSwitchToIdeFrame" to skip the iframe verification

File: tests/e2e/pageobjects/ide/Ide.ts [hunk 2]
```diff
@@ -63,6 +63,18 @@ export class Ide {
         }
     }
 
+    async verifyAndSwitchToIdeFrame(timeout: number = TimeoutConstants.TS_IDE_LOAD_TIMEOUT) {
```
Comment (ScrewTSW) [inline tests/e2e/pageobjects/ide/Ide.ts]: @makambalaji  This should be handled in code in `testsLibrary/ProjectAndFileTests.ts`, there's no point in adding this extra method.

File: tests/e2e/testsLibrary/WorkspaceHandlingTests.ts [hunk 3]
```diff
@@ -29,16 +30,37 @@ export class WorkspaceHandlingTests {
         @inject(CLASSES.Dashboard) private readonly dashboard: Dashboard,
         @inject(CLASSES.CreateWorkspace) private readonly createWorkspace: CreateWorkspace,
         @inject(CLASSES.Workspaces) private readonly workspaces: Workspaces,
-        @inject(CLASSES.WorkspaceNameHandler) private readonly workspaceNameHandler: WorkspaceNameHandler) {}
+        @inject(CLASSES.BrowserTabsUtil) private readonly browserTabsUtil: BrowserTabsUtil) {}
+
+    public getWorkSpaceName(stack: string) {
```
Comment (ScrewTSW) [inline tests/e2e/testsLibrary/WorkspaceHandlingTests.ts]: @makambalaji  This doesn't look like a safe method. will make it super confusing when the project name changes or new devfiles are added. please revert this `switch` case

File: tests/e2e/utils/requestHandlers/headers/CheMultiuserAuthorizationHeaderHandler.ts [hunk 4]
```diff
@@ -8,20 +8,14 @@
...
 @injectable()
 export class CheMultiuserAuthorizationHeaderHandler implements IAuthorizationHeaderHandler {
 
-    constructor(@inject(TYPES.ITokenHandler) private readonly tokenHandler: ITokenHandler) {
-    }
-
     async get(): Promise<AxiosRequestConfig> {
-        const token = await this.tokenHandler.get();
-        return { headers: { 'Authorization': `Bearer ${token}` } };
+        //  to-do : Fetch the cookies from user api and pass it here
```
Comment (ScrewTSW) [inline tests/e2e/utils/requestHandlers/headers/CheMultiuserAuthorizationHeaderHandler.ts:18]: @makambalaji  Do the tests still work without this logic in place? Are we sure we can leave it as WIP? Is there a follow-up issue?

**Review by ScrewTSW — CHANGES_REQUESTED** [review]

**Comment by ScrewTSW** [comment]
> Looks like iframe tag is removed in the Eclipse Che 7.43 version, So created a separate function "verifyAndSwitchToIdeFrame" to skip the iframe verification

Are you sure the CSS just didn't change?

File: tests/e2e/driver/CheReporter.ts (hunk 1 above)
Comment (makambalaji) [reply tests/e2e/driver/CheReporter.ts:89]: [preferencesHandler methods](https://github.com/eclipse/che/blob/main/tests/e2e/utils/PreferencesHandler.ts#L94-L122) are depends on the API, Currently API bearer token is not getting generated. So commented all those specific functions

File: tests/e2e/utils/requestHandlers/headers/CheMultiuserAuthorizationHeaderHandler.ts (hunk 4 above)
Comment (makambalaji) [reply tests/e2e/utils/requestHandlers/headers/CheMultiuserAuthorizationHeaderHandler.ts:18]: This token is used mainly for the api related functions [Eg: PreferenceHandler functions], which are not using in spec files any more. so it won't be any issue
Follow up issue : https://github.com/eclipse/che/issues/21203

File: tests/e2e/pageobjects/ide/Ide.ts (hunk 2 above)
Comment (makambalaji) [reply tests/e2e/pageobjects/ide/Ide.ts]: In multiple files, using the same logic, So IMO, its good to keep this function

File: tests/e2e/driver/CheReporter.ts (hunk 1 above)
Comment (ScrewTSW) [reply tests/e2e/driver/CheReporter.ts:89]: Ok, in that case that's not good. Do you know how are the API requests authorized without dedicated keycloak? These requests are CRITICAL for correct execution of tests.

File: tests/e2e/pageobjects/ide/Ide.ts (hunk 2 above)
Comment (ScrewTSW) [reply tests/e2e/pageobjects/ide/Ide.ts]: I can only see the method called at one place in your PR.

File: tests/e2e/utils/requestHandlers/headers/CheMultiuserAuthorizationHeaderHandler.ts (hunk 4 above)
Comment (ScrewTSW) [reply tests/e2e/utils/requestHandlers/headers/CheMultiuserAuthorizationHeaderHandler.ts:18]: hmm, refer to the previous comment

File: tests/e2e/pageobjects/ide/Ide.ts (hunk 2 above)
Comment (makambalaji) [reply tests/e2e/pageobjects/ide/Ide.ts]: tests/e2e/pageobjects/ide/ProjectTree.ts, I used twice, For other spec file fixes. we can reuse this function

**Comment by makambalaji** [comment]
> > Looks like iframe tag is removed in the Eclipse Che 7.43 version, So created a separate function "verifyAndSwitchToIdeFrame" to skip the iframe verification
> 
> Are you sure the CSS didn't change instead?

Yes, Iframes are removed, verified both in Eclipse Che 7.43 version and CRW 2.16 version

File: tests/e2e/driver/CheReporter.ts (hunk 1 above)
Comment (makambalaji) [reply tests/e2e/driver/CheReporter.ts:89]: There is a possibility for authentication with Oauth_Proxy cookie
After immediate login, we can see this in the Request headers. Need to check from where we can fetch these details [copying from slack conversation], will work on this solution as part of github issue : https://github.com/eclipse/che/issues/21203

File: tests/e2e/testsLibrary/WorkspaceHandlingTests.ts (hunk 3 above)
Comment (makambalaji) [reply tests/e2e/testsLibrary/WorkspaceHandlingTests.ts]: Not using this code, any more.. Alternative is already added,

Comment (ScrewTSW) [reply tests/e2e/testsLibrary/WorkspaceHandlingTests.ts]: could you please remove the redundant code then?

Comment (makambalaji) [reply tests/e2e/testsLibrary/WorkspaceHandlingTests.ts]: Removed the redundant code

**Review by ScrewTSW — APPROVED** [review]
LGTM

**Pull request merged**